src/Compilers/CSharp/Portable/Symbols/Source/SourceIncompleteMemberFieldSymbol.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Text;
+using System.Threading;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// This symbol is used for incomplete members, for example, `public class A { public B }`
+    /// </summary>
+    internal sealed class SourceIncompleteMemberFieldSymbol : SourceMemberFieldSymbol
```

Comment:
>SourceIncompleteMemberFieldSymbol [](http://example.com/codeflow?start=26&length=33)

Given the current usage of ```IncompleteMemberSyntax``` in the compiler, it feels like there is no longer an advantage in keeping producing it in the parser. We could just switch to producing ```FieldDeclarationSyntax``` nodes with certain parts as missing. Then we wouldn't need to add a special symbol type to support it.

Reply:
Will we remove `IncompleteMemberSyntax` completely? That feels like a large breaking change.

Reply:
> Will we remove IncompleteMemberSyntax completely? That feels like a large breaking change.

We don't have to remove the type right away. We could simply start with adding a comment saying the node is no longer produced. In the next release we could place a "warning" obsolete attribute on it, then change it to error, then remove the type at some point.

Reply:
@AlekseyTs This can also start producing NREs in places where it's assumed that `fieldDecl.Declaration.Type` is non-null.

This can possibly cause many analyzers out there to crash for this error scenario.

Is that an acceptable breaking change too?

Reply:
> This can also start producing NREs in places where it's assumed that fieldDecl.Declaration.Type is non-null.

I am not sure why that would be the necessary outcome. As I mentioned earlier:

> We could just switch to producing FieldDeclarationSyntax nodes with certain parts as missing.

Missing in this context is not necessarily null. For example when type is missing in other contexts where type is required, we create a missing identifier token, etc.

Reply:
> Missing in this context is not necessarily null.

Thanks for clarifying. I'll look into the suggested change.

Reply:
Would we want to make a similar change for VB as well?

Reply:
@333fred Yeah both languages will have similar changes.

Reply:
> In the next release we could place a "warning" obsolete attribute on it

I added the attribute now in this PR **only** for the purpose of revising our usages of it. I'll remove it before merging and open a tracking issue to add it back in a next release.